src/core/BufferAttribute.js

```diff
@@ -27,6 +26,8 @@ class BufferAttribute {
 		this.count = array !== undefined ? array.length / itemSize : 0;
 		this.normalized = normalized === true;
 
+		this.type = getArrayType( array );
+
```

Comment:
What's the value of this field?

Reply:
Is it possible to avoid this property and just evaluate the typed array in `normalize()`/`denormalize()`?

Reply:
I think passing the entire array into normalize/denormalize as a huge 'type' argument was never ideal as a public MathUtils API... if this went back into MathUtils at some point, I think it is better to use types like IntType / ShortType. But since this is just internal, I suppose it does not matter so much. Any preference between:

- (A) limit visibility of the property, e.g. `__type` or `#type`
- (B) continue passing the array 

/cc @WestLangley

Reply:
I prefer B for now until there's a practical reason to change it. It makes it hard to see what the other meaningful changes are in this PR, as well

Reply:
TBH, I prefer any form of (A). Passing the array is horrible, IMO.

Reply:
imo passing a three.js enum can result in unnecessarily verbose user code with no value. If something like "A" is used I'd prefer to pass in something like the array type / constructor: `normalize( value, Int8Array )`.

A lot of the code I've been working with has been written to generally support array types within three. For example packing buffers into textures for three-gpu-pathtracer, generating data structures in three-mesh-bvh, etc. A lot of the code in three.js isn't designed to support these types of use cases which is fine but it would be nice to keep it in mind when adding new functionality to the system - these use cases have been the driver for #24515.

Reply:
What if we keep these functions accessible through THREE.MathUtils, but rename them much more explicitly, and export getArrayType(array) as well. Then usage might look something like:

```javascript
import { MathUtils } from 'three';

const i = 127;
const type = MathUtils.getArrayType( Uint8Array );
const f = MathUtils.decodeNormalizedInt( i, type );
const i2 = MathUtils.encodeNormalizedInt( f, type );
```

I think this naming might be less likely to confuse people, as it's a more explicit about the use of integers as an encoding of floating-point values, per the [OpenGL definition](https://www.khronos.org/opengl/wiki/Normalized_Integer):

> A Normalized Integer is an integer which is used to store a decimal floating point number...

Reply:
I still not fully understanding the need for a separate set of "type" constants. I guess I'm just not understanding the need to redefine constants that the language already has built in with the typed array constructors. It just feels a bit roundabout and redundant - similar to the `Float32BufferAttribute`, `Uint8BufferAttribute`, etc classes. But I won't fight it if it's what the project wants.